### DISCUSSION TITLE
Can you tell me how to return the results of a test as a JSON file?

### DISCUSSION THREAD

**Discussion by sinyoung-park** [open-mmlab/mmclassification#1484] [Q&A] [answered]
I've completed CNN code in the past. However, I understand that pkl files are currently the only file type that can be returned via test.py. I would like to have the results in the form of a json file. How can I do this?

**Comment by Ezra-Yu** [comment, marked as answer]
Here is an example:

I get the result from the shell:
```
python tools/test.py $CFG $CKPT_PATH --out result.pkl --out-item pred
```

![image](https://user-images.githubusercontent.com/18586273/231931322-97a374c6-c3a6-4615-af5d-6e076cf9de46.png)


Note that:

result in pkl file is a list of dict:
each dict contains :('img_path', <class 'str'>), ('scale_factor', <class 'tuple'>), ('ori_shape', <class 'tuple'>), ('img_shape', <class 'tuple'>), ('sample_idx', <class 'int'>), ('num_classes', <class 'int'>), ('pred_score', <class 'torch.Tensor'>), ('pred_label', <class 'torch.Tensor'>), ('gt_score', <class 'torch.Tensor'>), ('gt_label', <class 'torch.Tensor'>)

make sure to change to `torch.Tensor` into python float, int, and list before dump into JSON file.

**Reply by sinyoung-park** [reply]
Thanks for the good solution! I solved this problem.
